### DISCUSSION TITLE
Need help with a RULE triggered by two conditions

### DISCUSSION THREAD

**Discussion by paggps** [arendst/Tasmota#22674] [Q&A] [answered]
Hello,

I am using Tasmota w/BLE, trying to receive iBeacons signals from an android phone with iBeacons generator.

I am receiving iBeacon events like this:

`23:52:11.406 RSL: SENSOR = {"Time":"2024-12-18T23:52:11","IBEACON":{"MAC":"5F64CA57CE43","NAME":"KNOWNPG","RSSI":-48,"STATE":"ON","PERSEC":1}}`

I am trying to create a rule that triggers when **IBEACON#NAME=KNOWNPG** and **IBEACON#STATE=ON**
So I wrote a rule like this:

`Rule1 ON IBEACON#NAME=KNOWNPG DO IF (IBEACON#STATE=ON) power1 toggle ENDIF ENDON`

But there is a problem (I don't understand where) with the checking of  (IBEACON#STATE=ON), because the Rule is triggered, but the action POWER1 TOGGLE is never executed.
If I replace the IF statement with IF (ON=ON), obviously the action is performed. So there is a problem on the IF (IBEACON#STATE=ON) part.

Thank you very much for any help!

**Comment by sfromis** [comment, marked as answer]
There is no straightforward way of doing that with rules, but since Bluetooth is typically on ESP32 devices, I'd recommend you to use the Berry language, where the Tasmota integration specifically allows combined trigger conditions. 

The issue with your attempt is that trigger expressions with `#` can only be used in the trigger part of a rule, before `do`, only a single trigger value, and not available in the body part after `do`. You'd have to use multiple rules, but again, Berry is a better approach.

**Reply by paggps** [reply]
Thank you very much for taking your time to answer. I will take a look to Berry (never used). Thanks.
